jiant/tasks/lib/superglue_axb.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+
+import jiant.tasks.lib.rte as rte
+
+
+@dataclass
+class Example(rte.Example):
+    pass
+
+
+@dataclass
+class TokenizedExample(rte.Example):
+    pass
+
+
+@dataclass
+class DataRow(rte.DataRow):
+    pass
+
+
+@dataclass
+class Batch(rte.Batch):
+    pass
+
+
+class SuperglueBroadcoverageDiagnosticsTask(rte.RteTask):
+    def get_train_examples(self):
+        return None
```

Comment:
raise NotImplementedError?

Reply:
I was debating this as well...For this task there are no train examples so returning `None` is appropriate behaviour

Reply:
Where this will likely get called is the `tokenize_and_cache` scripts, where the examples are passed through several functions before being iterated over. Having a fail-early NotImplemented error would be a more informative error than an iteration over None somewhere deeper in the call-stack (especially since I can see it being an easy mistake of calling `tokenize_and_cache` for train+val for all tasks being used.)

(Alternatively, we should probably add `get_train/test/val_examples` to the Task base class and have them raise by default unless overridden.)

Reply:
Would you be satisfied if I tested the pipeline for train/val on `None` for this task?

Reply:
I think it will throw an error in either case, just that it well get thrown deeper in the call-stack. Worth a quick test, but also I don't think it should be a blocker either way. Happy to approve all 3 PRs once verified either way.

Reply:
@zphang A little more detail on my reluctance to throw a `NotImplementedError()`
From the Python docs: https://docs.python.org/3/library/exceptions.html#NotImplementedError

Note It should not be used to indicate that an operator or method is not meant to be supported at all – in that case either leave the operator / method undefined or, if a subclass, set it to None.

As you expected, `tokenize_and_cache()` throws a fairly nondescript error. However, my solution to this is to gracefully handle an empty list shown here: https://github.com/jiant-dev/jiant/pull/115/commits/0c3cb2ff6448b00e49020adfe50804a21eb0da40

Reply:
In that case would it be better to just not have that method at all? I would prefer it not "fail" silently on a data phase that is not supported. I think in your proposed solution it will write out an empty train cache file, for instance.

Reply:
In my implementation, nothing is written to cache for train and validation (there are no files generated).

Reply:
I tried the version at https://github.com/jiant-dev/jiant/commit/46d948b749b4dfd18a604447b7320854eb1eca00, and the following command does write out a train folder with a `data_args.p`

```bash
python jiant/proj/main/tokenize_and_cache.py \
    --task_config_path ${BASE_PATH}/tasks/configs/glue_diagnostics_config.json \
    --model_type bert-base-uncased \
    --model_tokenizer_path /home/zp489/aliases/model_cache/tokenizers/bert-base-uncased/ \
    --phases "train" \
    --output_dir ${BASE_PATH}/cache/glue_diagnostics
```

Reply:
What is the preferred behavior in this scenario? No train folder created and no data_args.p created? No data chunks are created which is expected and desired.

Reply:
It should throw an error because there is no valid train phase for this task. Removing the `get_train/val_examples` methods would lead to an error that informs the user that the task has no `get_*_examples` implementations for those phases.

Reply:
If possible let's remove the none-handling as well. Ideally we don't want it to fail/write silently.